aws_protocol_adaptor/device_client/aws_nvmsgbroker.c

```diff
@@ -419,20 +419,22 @@ char *generate_sha256_hash(char *str)
 
 NvDsMsgApiErrorType nvds_msgapi_connection_signature(char *broker_str, char *cfg, char *output_str, int max_len)
 {
-	strcpy(output_str, "");
+	strcpy(output_str, ""); // Initializing output_str as empty string if successful which is updated later by SHA-256 hash else empty string is returned
```

Comment:
This comment isn't very helpful. 
```suggestion
	// Value of output_str must be empty string if operation is unsuccessful
	strcpy(output_str, "");
```